frontend/catalyst/jax_tracer.py

```diff
@@ -1386,11 +1388,16 @@ def is_leaf(obj):
                 device_shots = get_device_shots(device) or 0
                 device_init_p.bind(
                     device_shots,
+                    auto_qubit_management=(device.wires is None),
                     rtd_lib=device.backend_lib,
                     rtd_name=device.backend_name,
                     rtd_kwargs=str(device.backend_kwargs),
                 )
-                if catalyst.device.qjit_device.is_dynamic_wires(device.wires):
+                if device.wires is None:
+                    # Automatic qubit management mode
+                    # We start with 0 wires and allocate new wires in runtime as we encounter them.
+                    qreg_in = qalloc_p.bind(0)
+                elif catalyst.device.qjit_device.is_dynamic_wires(device.wires):
                     # When device has dynamic wires, the device.wires iterable object
                     # has a single value, which is the tracer for the number of wires
                     qreg_in = qalloc_p.bind(device.wires[0])
```

Comment:
I still find it strange how this branch differs here compared to the measurements, but oh well 😅